Shared/Flows/Onboarding/OnboardingCoordinator.swift

```diff
@@ -21,40 +21,142 @@ class OnboardingCoordinator: PresentableCoordinator<Void> {
     }
 
     override func start() {
+        self.setInitialOnboardingContent()
         self.handle(deeplink: self.deepLink)
     }
 
     func handle(deeplink: DeepLinkable?) {
         guard let link = deeplink else { return }
+
         self.onboardingVC.reservationId = link.reservationId
         self.onboardingVC.passId = link.passId
         self.onboardingVC.updateUI()
     }
-}
 
-extension OnboardingCoordinator: LaunchActivityHandler {
+    // MARK: - Onboarding Flow Logic
 
-    nonisolated func handle(launchActivity: LaunchActivity) {
-        Task.onMainActor {
-            self.onboardingVC.handle(launchActivity: launchActivity)
+    private func setInitialOnboardingContent() {
+        let userStatus = User.current()?.status
+
+        let initialContent: OnboardingContent
+        switch userStatus {
+        case .needsVerification, .inactive, .waitlist, .none:
+            initialContent = .welcome(self.onboardingVC.welcomeVC)
+        case .active:
+            self.finishFlow(with: ())
+            return
+        }
+
+        self.onboardingVC.switchTo(initialContent)
+    }
```

Comment:
Because the coordinator is responsible for flow, it also has to determine the initial onboarding content.